File: user_tests/simple_class_test/simple_class.py
class Person:
    def __init__(self, name:str):
        self.name = name

    def get_name(self):
        return self.name

class Student(Person):
    def __init__(self, 
                 name:str, 
                 student_number:int,
                 domain:str = "school.student.pt"):
        if student_number < 0:
            raise ValueError("Student number must be a positive number")
        self.student_number = student_number
        self.name = name

    def get_name(self):
        return f"{self.name} - {self.student_number}"

File: user_tests/simple_class_test/simple_class.jl

abstract type AbstractPerson end
abstract type AbstractStudent <: AbstractPerson end

mutable struct Person <: AbstractPerson
    name::String
end
function get_name(self::AbstractPerson)::String
    return self.name
end

mutable struct Student <: AbstractStudent
    name::String
    student_number::Int64
    domain::String

    Student(name::String, student_number::Int64, domain::String = "school.student.pt") =
        begin
            if student_number < 0
                throw(ValueError("Student number must be a positive number"))
            end
            new(name, student_number, domain)
        end
end